### ISSUE TITLE
short-circuit a pipeline? 

### ISSUE THREAD

**Issue by zzztimbo** [dagster-io/dagster#2176]
I have a pipeline that is supposed to process some data. 
Sometimes, I don't have any data to process. 

Is there a way to end my pipeline early? Currently, my plan is to configure my solids to expect Optional input. Is this a good design pattern?

**Comment by alangenfeld** [comment]
Yep! Assuming you are talking about using is_optional on OutputDefinition . If an optional output is not returned, all downstream solids that depend on it will skip . This should be a great way to achieve your goal of short-circuiting.

**Comment by zzztimbo** [comment]
thank you @alangenfeld 

this is a good example here: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster_tests/core_tests/execution_tests/test_retries.py#L18-L26

**Comment by zzztimbo** [comment]
@alangenfeld it seems that if I have another solid downstream of a skipped solid, I will get a pipeline failure. 

```
from dagster import Output
from dagster import OutputDefinition
from dagster import pipeline
from dagster import solid


@solid(
    output_defs=[
        OutputDefinition(bool, "start_fail", is_optional=True),
        OutputDefinition(bool, "start_skip", is_optional=True),
    ]
)
def two_outputs(_):
    yield Output(True, "start_fail")
    # won't yield start_skip


@solid(
   output_defs=[
       OutputDefinition(bool, "start_skip", is_optional=True),
   ]
)
def will_be_skipped(_, _start_skip):
    if _start_skip:
        yield Output(_start_skip, "start_skip")


@solid
def will_be_skipped_level_2(_, _start_skip):
    pass


@solid
def print_start_fail(context, sf):
    context.log.info(str(sf))


@pipeline
def pipe():
    start_fail, start_skip = two_outputs()
    print_start_fail(start_fail)
    skip_result = will_be_skipped(start_skip)
    will_be_skipped_level_2(skip_result)
```

**Comment by alangenfeld** [comment]
oof, thats not right. Let me get that bug fixed ASAP

**Comment by alangenfeld** [comment]
https://dagster.phacility.com/D2147
fix should go out in `0.7.2` today or tomorrow
